### PR TITLE
added comand-line argument parsing to compile_packages.py

### PR DESCRIPTION
**Pull request by Laudanum-dev**
added argparse to compile_packages.py. also rewrote a bit of a code in directory-searching functions + some code "reformat" by pycharm. now this utility is a bit better and easier to use .

changes displayed below are quite messy cause there have been some reformat by pycharm

### CODE REVIEW

File: compile_packages.py [hunk 1]
```diff
@@ -159,41 +165,129 @@ def build_package(package_path, pack_zip_path, zip_path):
...
+            else:
+                shutil.move("zips/lrg", output)
+            if not args.preserve_temp:
+                print("Deleting temporary files")
+                shutil.rmtree("zips")
+                print("Done!")
+        else:
+            if not args.preserve_temp:
+                print("Deleting temporary files")
+                shutil.rmtree("zips/hammer")
```
Comment (TeamSpen210) [inline compile_packages.py:288]: The `hammer/` folder isn't actually temporary, it's a dump of all the resources, so you could manually load them yourself. I was thinking about releasing that in addition to the packages for Hammer users.

Comment (Laudanum-dev) [reply compile_packages.py:288]: oh, thats cool, so maybe i should change text so it says "hammer files" instead of "temporary files"

Comment (TeamSpen210) [reply compile_packages.py:290]: I mean that it shouldn't delete them, maybe move it out of zips then.

Comment (Laudanum-dev) [reply compile_packages.py]: ok, i updated it a bit, better show what help says, 
image

Comment (Laudanum-dev) [reply compile_packages.py]: plus, it won't create any temporary folders, all folders it creates now are output folder and hammer folder, thats all 
only things that are getting deleted by now are unzipped .bee_pack's if an option to zip them all in one zip is chosen

**Pull request merged**